interfaces/builtin/anbox_support.go

```diff
@@ -0,0 +1,512 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const anboxSupportSummary = `allows operating as the Anbox container management service`
+
+const anboxSupportBaseDeclarationPlugs = `
+  anbox-support:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const anboxSupportBaseDeclarationSlots = `
+  anbox-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const anboxSupportConnectedPlugAppArmor = `
+/sys/module/fuse/parameters/userns_mounts rw,
+/run/anbox/{,**} rw,     # SNAP_COMMON?
+
+# aa-exec -p ...//lxc -- ...
+@{PROC}/[0-9]*/attr/exec w,
+@{PROC}/[0-9]*/attr/current r,
+
+change_profile unsafe /** -> @{profile_name}//lxc,
+profile lxc (attach_disconnected,mediate_deleted) {
+    file,
+    ptrace,
+    signal,
+    dbus,
+    capability,
+    mount,
+    umount,
+    remount,
+    network,
+
+    # snapctl and its requirements; needed as we're now in a child profile
+    /usr/bin/snapctl ixr,
+    @{PROC}/sys/net/core/somaxconn r,
+    /run/snapd-snap.socket rw,
+
+    #include <abstractions/base>
+    #include <abstractions/nameservice>
+    /dev/tty rw,
+    @{INSTALL_DIR}/@{SNAP_NAME}/{,**} mrix,  # fine-tune
+    /var/snap/@{SNAP_NAME}/{,**} rwk,        # fine-tune
+    /{,usr/}{,s}bin/* ixr,                   # fine-tune
+
+    @{PROC}/[0-9]*/cgroup r,
+    @{PROC}/[0-9]*/mounts r,
+    @{PROC}/[0-9]*/mountinfo r,
+    /sys/module/apparmor/parameters/enabled r,
+
+    # KERNEL=="loop-control" in udev connected plug
+    /dev/loop-control rw,
+    # KERNEL=="loop*", SUBSYSTEM=="block" in udev connected plug
+    /dev/loop[0-9]* rw,
+
+    /sys/fs/cgroup/unified/cgroup.controllers r,  # checking if using unified?
+
+    # Anbox comes with a squashfs image which contains the entire Android rootfs which
+    # has to be mounted in place together with the read-write directories data and cache.
+    mount fstype=squashfs options=(ro) /dev/loop[0-9]* -> /var/snap/@{SNAP_NAME}/common/rootfs/,
+    mount options=(rw, bind) /var/snap/@{SNAP_NAME}/common/cache/ -> /var/snap/@{SNAP_NAME}/common/rootfs/cache/,
+    mount options=(rw, bind) /var/snap/@{SNAP_NAME}/common/data/ -> /var/snap/@{SNAP_NAME}/common/rootfs/data/,
+    umount /var/snap/@{SNAP_NAME}/common/rootfs/,
+    umount /var/snap/@{SNAP_NAME}/common/rootfs/cache/,
+    umount /var/snap/@{SNAP_NAME}/common/rootfs/data/,
+
```

Comment:
We need to handle the overlayfs feature Anbox supports as well. See https://github.com/anbox/anbox/blob/master/src/anbox/cmds/container_manager.cpp#L233 and https://docs.anbox.io/userguide/advanced/rootfs_overlay.html